productcomments.php

```diff
@@ -922,65 +923,54 @@ private function renderProductCommentModal($product)
-        ));
-
-        return $this->context->smarty->fetch('module:productcomments/views/templates/hook/product-list-reviews.tpl');
+            'nb_comments' => $commentsNb,
+            'post_allowed' => $isPostAllowed,
+        );
     }
 
-    /**
-     * Display average grade and buttons in the product page under checkout and share buttons
-     *
-     * @param $params
-     *
-     * @return string
-     *
-     * @throws PrestaShopException
-     * @throws SmartyException
-     */
-    public function hookDisplayProductAdditionalInfo($params)
+
+    public function renderWidget($hookName = null, array $configuration = [])
     {
-        /** @var ProductLazyArray $product */
-        $product = $params['product'];
-        /** @var ProductCommentRepository $productCommentRepository */
-        $productCommentRepository = $this->context->controller->getContainer()->get('product_comment_repository');
 
-        $averageGrade = $productCommentRepository->getAverageGrade($product->getId(), Configuration::get('PRODUCT_COMMENTS_MODERATE'));
-        $commentsNb = $productCommentRepository->getCommentsNumber($product->getId(), Configuration::get('PRODUCT_COMMENTS_MODERATE'));
-        $isPostAllowed = $productCommentRepository->isPostAllowed($product->getId(), (int) $this->context->cookie->id_customer, (int) $this->context->cookie->id_guest);
+        if('displayProductListReviews' === $hookName || isset($configuration['type']) && 'product_list' === $configuration['type']) {
 
-        $this->context->smarty->assign(array(
-            'average_grade' => $averageGrade,
-            'nb_comments' => $commentsNb,
-            'post_allowed' => $isPostAllowed,
-        ));
+            $product = $configuration['product'];
+            $id_product = $product['id_product'];
+            $variables = $this->getWidgetVariables($hookName, array('id_product' => $id_product));
+
+            $variables = array_merge($variables, array(
+                'product' => $product,
+                'product_comment_grade_url' => $this->context->link->getModuleLink('productcomments', 'CommentGrade')
+            ));
+
+            $file_path = 'module:productcomments/views/templates/hook/product-list-reviews.tpl';
+
+        } elseif ($this->context->controller instanceof ProductControllerCore) {
 
-        if ('quickview' === Tools::getValue('action')) {
-            return $this->context->smarty->fetch('module:productcomments/views/templates/hook/product-additional-info-quickview.tpl');
+            $id_product = Tools::getValue('id_product');
+            $variables = $this->getWidgetVariables($hookName, array('id_product' => $id_product));
+
+            $file_path = 'quickview' === Tools::getValue('action') ? 'module:productcomments/views/templates/hook/product-additional-info-quickview.tpl' : 'module:productcomments/views/templates/hook/product-additional-info.tpl';
+    
+        }
+
+        if (empty($variables)) {
```

Comment:
I understand that if we dont get into the IF or ELSEIF $variables is undefined. Can you initialize it with $variables = [] before the conditions ?